### DISCUSSION TITLE
Ability to toggle "Create New" and "Add Existing" buttons for relational fields as interface options

### DISCUSSION THREAD

**Discussion by jpetryk** [directus/directus#5218] [Feature Requests]
It would be nice to be able to control whether or not these buttons appear as there are use cases in which you want to force users to only create new items or only choose from existing items. This likely varies depending on the collection or even fields within a collection, but this could be accomplished via CSS if forms were wrapped in a class that reflecting the collection or field name, but this is a bit hacky even if was possible, and certainly not that easily maintained over time.

Best case scenario is that they are simply options within the interfaces these apply to. i.e., "Hide Create New button" and "Hide Add Existing button".

This has been identified as a potential oversight of implementing https://github.com/directus/directus/issues/3647, which is now closed, but was one of the considerations of that issue. It is also separate from https://github.com/directus/directus/issues/4192 which would be to control the display of these buttons through permissions. Also potentially a useful feature, but a different use case.

**Comment by rijkvanzanten** [comment]
I'm wondering if this should be based on permissions* or an option in the interface 🤔 

\* Not permissions to the related item, but permissions to create items _through_ the relationship. Your use case

> as there are use cases in which you want to force users to only create new items or only choose from existing items

sounds more like a permission than a visual tweak to me (the downside with doing it as an interface option is that the user could still hack around it through the API directly)

**Reply by jpetryk** [reply]
As long as each could be toggled separately it satisfies my use case. I'm wondering the form this takes within the Field Permissions area since both buttons essentially are "Adding". In one case you are creating an item in another collection and then adding, or you are choosing from items in another collection and adding.

So... if we're doing this through the underlying collection defining the relationship:

Create = yes, Edit = no, View = no, Delete = no -> results in "Add New" showing, "Add Existing" hidden
Create = yes, Edit = no, View = yes, Delete = no -> results in "Add New" hidden, "Add Existing" showing

Something like that?

**Reply by benhaynes** [reply]
In addition to permissions, I think we should have two interface options to hide these buttons individually. It will be cumbersome to configure this in permissions, and as long as we also enforce permissions, this will give us the best of both worlds.

**Reply by jpetryk** [reply]
I would agree with this because if it's the same for every user role, which I would think might be a more common use case given that this is likely a content-driven implementation (type of content we're representing) that would require said feature rather than a use case in which different users need to see different buttons.

**Reply by jpetryk** [reply]
I also noticed my formula for the permission sets capturing the 2 scenarios don't totally make sense (there's crossover). I was just spit balling there.

**Reply by fballan93** [reply]
funny thing is i actually went inside the discussion section to write  this exactly . for us currently we are forcing to hide it for all by removing the css and removing the html element from the page on open for a custom role. having this nativity is a big plus for sure .

**Comment by jpetryk** [comment]
To clarify this issue: this functionality already exists for O2M relations via the Interface options "Enable Create Button" and "Enable Select Button". So there's already a standard set here.

For consistency, it makes sense to add the same Interface options to M2A, M2M and M2O.

For M2O, the effect would be on the + and arrow used to add or select existing (or again, for consistency, those change to buttons like the other relations types).

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
